docs/modules/reference/pages/resources.adoc

```diff
@@ -0,0 +1,137 @@
+            [io.pedestal.http.route :as route]))
+
+(defn create-server
+  []
+  (http/create-server
+    {::http/port 8890
+     ::http/type :jetty
+     ::http/join? false
+     ::http/routes (route/routes-from
+                     #{ ... }  <1>
+                     (resources/file-routes {:file-root "web/public"}))})) <2>
+----
+
+<1> This is where you would put the dynamic routes for your application.
+<2> This exposes any files stored in the `web/public` folder under the default URI path `/public`.
+
+The `routes-from` macro will expand and combine the routes together to form the full routing table.
+
+The api:resource-routes[] function is analagous to `file-routes`, but returns routes that expose resources on the classpath.
+
+Each of these functions take a single parameter, a map of options:
+
+.Options to file and resource functions
+[options=header,cols="1,1,^1,2p"]
+|===
+| Option | Default | Only | Description
+
+| :prefix          | "/public" |          | The path prefix for the exposed resources.
+| :root-path       | -         | file     | Path to root directory to expose.
+| :resource-root   | -         | resource | Prefix on resources, limiting access to a safe subset of the classpath
+| :classloader     | -         | resource | Classloader used to find resources, or nil for the default class loader
+| :index-files?    | true      | file     | If true, paths that map to directories may return the index file for that directory.
+| :cache?          | true      |          | If true, then resource data is cached to in memory to speed up requests.
+| :fast?           | true      |          | If true, then fast, asynchronous responses may be used.
+| :allow-head?     | true      |          | If true, then additional routes are added to support the :head request method,
+                                            returning empty bodies.
+| :route-namespace | "io.pedestal.http.resources"
+                               |          | Namespace used when creating route names for the routes.
+|===
+
+Using the defaults, which enable fast responses and caching, the route-based resources are demonstrably faster
+than the xref:#service-map-options[interceptor-based] resources, especially for large resources ... though
+for any reasonably sized resource, the response time is sub-millisecond.
+
+=== Options notes
+
+It is quite possible to use both `file` and `resource` together, as long as the :prefix is set so that routes do not overlap.
+If necessary, it's even possible to use these functions multiple times, but care must be taken with the :route-namespace to avoid
+route name collisions.
+
+The :prefix option should start with a "/" and not end with one.
+
+For the :index-files? option, Pedestal will identify requests that map to a directory, not to a specific file.
+Pedestal will search for one of the following:
+
+- `index.html`
+- `index.htm`
+- Any file whose name starts with `index.`
```

Comment:
We could make this a numbered list if it tries to find these files in the order (which I assume it does; if that is not the case, we should clarify)